lib/valueflow.cpp

```diff
@@ -206,6 +206,12 @@ static void setConditionalValues(const Token* tok,
 {
     if (Token::Match(tok, "==|!=|>=|<=")) {
         true_value = ValueFlow::Value{tok, value};
+        const Variable* var1 = tok->astOperand1()->variable();
+        const Variable* var2 = tok->astOperand2()->variable();
+        if ((var1 && var1->isFloatingType()) || (var2 && var2->isFloatingType())) {
+            true_value.valueType = ValueFlow::Value::ValueType::FLOAT;
+            false_value.valueType = ValueFlow::Value::ValueType::FLOAT;
+        }
```

Comment:
The code below doesnt work for floating point. This function never sets `floatValue` which is what the `ValueType::FLOAT`, and it uses incorrect logic for the bounds(ie `value + 1` instead of using float epsilon). It probably works because `floatValue` is zero by default, and you are only testing with equality.

Actually, this function is mainly used in `parseCompareInt`(and also for symbolic conditions) which are for integral-like comparisons.  I think it would be better to return `null` in `parseCompareInt` if the return token is a float. So you could add a check in addition to `isSaturated` check:

```cpp
const Token *parseCompareInt(const Token *tok, ValueFlow::Value &true_value, ValueFlow::Value &false_value, const std::function<std::vector<MathLib::bigint>(const Token*)>& evaluate)
{
    ...
    if (tok->isComparisonOp()) {
        ...
        if (!value1.empty()) {
            if (isSaturated(value1.front()) || astIsFloat(tok->astOperand2()))
                return nullptr;
            setConditionalValues(tok, true, value1.front(), true_value, false_value);
            return tok->astOperand2();
        } else if (!value2.empty()) {
            if (isSaturated(value2.front())|| astIsFloat(tok->astOperand1()))
                return nullptr;
            setConditionalValues(tok, false, value2.front(), true_value, false_value);
            return tok->astOperand1();
        }
    }
    ...
}
```

We should probably add better support for float comparisons in the future, but they are little difficult due to not being totally ordered(ie `d < 0.0 != 0.0 > d`).